package.json

```diff
@@ -45,15 +45,15 @@
     "bindings": "^1.5.0",
     "eventemitter2": "^5.0.1",
     "nan": "^2.15.0",
-    "prebuild-install": "^6.1.4"
+    "prebuild-install": "^7.0.1"
   },
   "devDependencies": {
     "chai": "^4.1.2",
     "chalk": "^2.4.1",
     "eslint": "^4.19.1",
     "jasmine": "^3.1.0",
-    "node-abi": "^3.1.0",
-    "node-gyp": "^8.2.0",
```

Comment:
Why is it trying to use `gyp info using node-gyp@5.1.0` in the actual build? https://github.com/MadLittleMods/node-usb-detection/runs/5399319750?check_suite_focus=true

Must be what's installed with the Node.js version, `C:\\hostedtoolcache\\windows\\node\\14.19.0\\x64\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js`

Reply:
Looks like it might be connected to the npm version, https://github.com/nodejs/node-gyp/blob/master/docs/Updating-npm-bundled-node-gyp.md

Reply:
Based on those documents, I tried many different ways to update `node-gyp` and while it works for most platforms, I couldn't get it to work on Windows with `npm@6`. It always used `node-gyp@5.1.0` no matter what (expected `node-gyp@8.4.0`).


```
      # Make sure the build is using the latest version of node-gyp (which also
      # matches our package.json) to ensure maximum build compatibility and
      # avoid random build errors.
      #
      # See
      # - https://github.com/nodejs/node-gyp/blob/245cd5bbe4441d4f05e88f2fa20a86425419b6af/docs/Updating-npm-bundled-node-gyp.md
      # - https://github.com/nodejs/node-gyp/blob/245cd5bbe4441d4f05e88f2fa20a86425419b6af/docs/Force-npm-to-use-global-node-gyp.md
      #
      # For macOS/Linux and npm <= 6
      - name: 'Install the latest version of node-gyp@^8'
        if: ${{ matrix.os != 'Windows' && (matrix.node == '12' || matrix.node == '14') }}
        run: npm explore npm/node_modules/npm-lifecycle -g -- npm install node-gyp@^8
      # For macOS/Linux and npm >= 7
      - name: 'Install the latest version of node-gyp@^8'
        if: ${{ matrix.os != 'Windows' && (matrix.node == '16') }}
        run: npm explore npm/node_modules/@npmcli/run-script -g -- npm_config_global=false npm install node-gyp@^8
      # For Windows and npm <= 6
      - name: 'Install the latest version of node-gyp@^8'
        if: ${{ matrix.os == 'Windows' && (matrix.node == '12' || matrix.node == '14') }}
        run: |
          "@FOR /f \"tokens=*\" %i IN ('where node') DO cd/d \"%~dpi\""
          cd node_modules\\npm\\node_modules\\npm-lifecycle
          npm install node-gyp@^8
          npm install --global node-gyp@^8
          for /f "delims=" %P in ('npm prefix -g') do npm config set node_gyp "%P\node_modules\node-gyp\bin\node-gyp.js"
        shell: cmd
      # For Windows and npm >= 7
      - name: 'Install the latest version of node-gyp@^8'
        if: ${{ matrix.os == 'Windows' && (matrix.node == '16') }}
        run: |
          "@FOR /f \"tokens=*\" %i IN ('where node') DO cd/d \"%~dpi\""
          cd node_modules\\npm\\node_modules\\@npmcli\\run-script
          npm install node-gyp@^8
        shell: cmd
```

Reply:
I resorted to just install `npm@^8` for all versions which has a newer bundled `node-gyp`. Still unclear how that's different from above though 🤷‍♀️

Reply:
Looks like someone saw this PR and solved it a different way: https://github.com/snowtrack/snowfs/pull/258